Orange/widgets/regression/owlinearregression.py

```diff
@@ -145,52 +132,38 @@ def commit(self):
         elif self.reg_type == OWLinearRegression.Elastic:
             learner = ElasticNetLearner(alpha=alpha,
                                         l1_ratio=self.l1_ratio, **args)
+        return learner
 
-        learner.name = self.learner_name
-        predictor = None
+    def update_model(self):
+        super().update_model()
         coef_table = None
-
-        self.error(0)
-        if self.data is not None:
-            if not learner.check_learner_adequacy(self.data.domain):
-                self.error(0, learner.learner_adequacy_err_msg)
-            else:
-                predictor = learner(self.data)
-                predictor.name = self.learner_name
-                domain = Domain(
+        if self.good_data:
+            domain = Domain(
                     [ContinuousVariable("coef", number_of_decimals=7)],
                     metas=[StringVariable("name")])
-                coefs = [predictor.intercept] + list(predictor.coefficients)
-                names = ["intercept"] + \
-                    [attr.name for attr in predictor.domain.attributes]
-                coef_table = Table(domain, list(zip(coefs, names)))
-                coef_table.name = "coefficients"
-
-        self.send("Linear Regression", learner)
-        self.send("Model", predictor)
+            coefs = [self.model.intercept] + list(self.model.coefficients)
+            names = ["intercept"] + \
+                    [attr.name for attr in self.model.domain.attributes]
+            coef_table = Table(domain, list(zip(coefs, names)))
```

Comment:
`domain` above has a single variable named 'coef' and one meta, 'name'. Does this line work?